storage/client_test.go

```diff
@@ -498,7 +498,6 @@ func (t *multiTestContextKVTransport) SendNext(done chan<- kv.BatchCall) {
 		done <- kv.BatchCall{Reply: br, Err: nil}
 	}) != nil {
 		done <- kv.BatchCall{Err: roachpb.NewSendError("store is stopped")}
-		t.mtc.expireLeases()
```

Comment:
Understood about the manual clock. If a store is stopped, `DistSender` will send a request to another store (if one is available) and we'll get a `NotLeaseHolderError` which will expire the lease. The problem with expiring leases in this code path is that the down store may not be the lease holder. In the case of `TestConcurrentRaftSnapshots`, we replica a range to stores 1, 2, 3, 4, 5, then we stop stores 2 and 3. If a request accidentally gets directed at one of the down stores then we expire leases and any subsequent request to store 1 (which is how we direct requests in most `multiTestContext`-based tests) will die with a `NotLeaseHolderError`. 

Note that we almost never stop store 1 in `multiTestContext`-based tests. One test which does is `TestRemoveRangeWithoutGC`. I've stressed that test with this PR and it appears ok.